frictionless_ckan_mapper/ckan_to_frictionless.py

```diff
@@ -150,14 +150,49 @@ def dataset(ckandict):
     # package_show can have license_id and license_title
     # TODO: do we always license_id i.e. can we have license_title w/o
     # license_id?
-    if ('licenses' not in outdict and 'license_id' in outdict):
-        outdict['licenses'] = [{
-            'name': outdict['license_id'],
-        }]
-        if 'license_title' in outdict:
-            outdict['licenses'][0]['title'] = outdict['license_title']
+    # There's no mention in the docs of license_title nor license_url, plus
+    # license_id is listed as optional in both CKAN 2.8 and 2.9.
+
+    # Looping like this because all those keys are optional.
```

Comment:
You should wrap this whole block in an if statement to check we don't have licenses already via extras (in which case we just use that ...

Reply:
This wouldn't apply if my idea is good enough (from this comment: https://github.com/frictionlessdata/frictionless-ckan-mapper/pull/32/files#r446375072) because `outdict['extras']` do not exist anymore at this point and 'licenses' can potentially be overwritten and repopulated depending on if we have one of 'license_id', 'license_title', 'license_url'.